src/Internal/Service/ExampleServiceWithDependencies.php

```diff
@@ -57,9 +58,17 @@ public function __construct(
 	/**
 	 * Temporary method to test the Mode.
 	 *
+	 * @throws Exception In case mode is not initialized yet.
 	 * @return bool
 	 */
 	public function is_in_test_mode() {
+		try {
+			$this->example_service->do_something_with_exception();
+		} catch ( Exception $e ) {
+			// Do nothing, this is fine. We're just testing Psalm.
+			return false;
+		}
+
```

Comment:
I understand this is just for testing Psalm, but for which purpose? 

- With it, `return $this->mode->is_test();` is never reached. 
- And without it, Psalm catches up correctly that this method needs to declare `@throw`.

Reply:
It's just an example, which ensures that Psalm reads both `src` and `includes`, and understands where exceptions might be thrown. The purpose here isn't valid code, I just wanted to make sure that there are a couple of lines, which when commented out, would yield something like this:

<img width="864" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/5311119/5c67bb77-598a-46ad-8acc-f826fc471939">

Of course, I cannot commit without the `@throws` and `try-catch`, because all checks would be 🔴 .